library/src/main/java/com/datatorrent/lib/io/fs/FileSplitter.java

```diff
@@ -99,6 +105,8 @@
 
   protected final BasicCounters<MutableLong> fileCounters;
 
+  private final static Lock LOCK = new Lock();
```

Comment:
Later if we want to enhance the lock then we will be able to do so. With Object we can't